windows/win_host.ps1

```diff
@@ -0,0 +1,277 @@
+    }
+
+    If ($params.domain) {
+        $domain = $params.domain.toString()
+        Set-Attr $result.win_host "domain" $domain
+        If ($domain -eq (gwmi WIN32_ComputerSystem).Domain) {
+            Exit-Json $result "The computer is already apart of the domain: $domain."
+        }
+        $domain = "-DomainName '$domain'"
+
+
+        If ($params.server) {
+            $server = $params.server.toString()
+            Set-Attr $result.win_host "server" $server
+            $server = "-Server '$server'"
+        }
+        Else {
+            $server = ""
+        }
+
+        If ($params.options) {
+            $options = $params.options.toString()
+            Set-Attr $result.win_host "options" $options
+            $options = "-Options '$options'"
+        }
+        Else {
+            $options = ""
+        }
+
+        If ($params.oupath) {
+            $oupath = $params.oupath.toString()
+            Set-Attr $result.win_host "oupath" $oupath
+            $oupath = "-OUPath '$oupath'"
+        }
+        Else {
+            $oupath = ""
+        }
+
+        If (($params.unsecure -eq "true") -or ($params.unsecure -eq "yes")) {
+            $unsecure = $params.unsecure.toString()
+            Set-Attr $result.win_host "unsecure" $unsecure
+            $unsecure = "-Unsecure"
+        }
+        Else {
+            $unsecure = ""
+        }
+    }
+    Else {
+        $domain = ""
+    }
+
+    If ($params.user -and $params.pass) {
+        $user = $params.user.toString()
+        $pass = $params.pass.toString()
+        $credential = "-Credential"
+        $unjoincredential = "-UnjoinDomainCredential"
+        $local = "-LocalCredential"
+
+        $creds = $true
+    }
+
+    If ($params.workgroup) {
+        $workgroup = $params.workgroup.toString()
+        Set-Attr $result.win_host "workgroup" $workgroup
+        $workgroup = "-WorkgroupName '$workgroup'"
+    }
+    Else {
+        If ($creds) {
+            Set-Attr $result.win_host "workgroup" "WORKGROUP"
+            $workgroup = "-WorkgroupName 'WORKGROUP'"
+        }
+        Else {
+            $workgroup = ""
+        }
+    }
+
+    If (($params.restart -eq "true") -or ($params.restart -eq "yes")) {
+        $restart = $true
+        Set-Attr $result.win_host "restart" "true"
+    }
+    Else {
+        Set-Attr $result.win_host "restart" "false"
+        $restart = $false
+    }
+
+    If ($params.state -eq "present") {
+        $state = $true
+        Set-Attr $result.win_host "state" "present"
+    }
+    ElseIf ($params.state -eq "absent") {
+        $state = $false
+        Set-Attr $result.win_host "state" "absent"
+    }
+    Else {
+        $state = "none"
+    }
+
+    # If just hostname was provided and not credentials and there was only one hostname just rename computer
+    If ($hostname -and -Not ($credential) -and -Not ($domain -Or $workgroup) -and $hostname.length -eq 1) {
+        Rename-Computer $hostname[0]
```

Comment:
You should have an If here so that the hostname is only changed when necessary.

E.g.
```python
        If (([System.Net.Dns]::GetHostName()).CompareTo($hostname) -ne 0)
        {
            Rename-Computer $hostname[0]
            $result.changed = $true
        }
```